### PR TITLE
Issues/86577 reland

### PR DESCRIPTION
**Pull request by chunhtai**
https://github.com/flutter/flutter/issues/86577

The previous pr causes post submit failure because the NSString is not initialize correctly, probably because MRC. This pr explicitly set it to nil. The change is in the second commit

## Pre-launch Checklist

- [ ] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I signed the [CLA].
- [ ] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/master/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: shell/platform/darwin/ios/framework/Source/SemanticsObject.mm [hunk 1]
```diff
@@ -610,14 +610,13 @@ - (BOOL)onCustomAccessibilityAction:(FlutterCustomAccessibilityAction*)action {
...
-  NSString* label;
+  NSString* label = nil;
   if (![self node].label.empty()) {
     label = @([self node].label.data());
   }
   if (![self node].tooltip.empty()) {
-    label = [NSString stringWithFormat:@"%@%@",
-                                       label ? [NSString stringWithFormat:@"%@\n", label] : @"",
-                                       @([self node].tooltip.data())];
+    label = label ? [NSString stringWithFormat:@"%@\n%@", label, @([self node].tooltip.data())]
```
Comment (chunhtai) [inline shell/platform/darwin/ios/framework/Source/SemanticsObject.mm:618]: same logic but cleaner

Comment (blasten) [reply shell/platform/darwin/ios/framework/Source/SemanticsObject.mm:618]: cc @jmagman for readability

Comment (jmagman) [reply shell/platform/darwin/ios/framework/Source/SemanticsObject.mm:618]: You could also use a `NSMutableString` to build up the string.  This works too, though.

**Review by jmagman — APPROVED** [review]
Why didn't the original code produce a `Undefined or garbage value returned to caller` or an uninitialized static analyzer warning?  That's concerning...

Should be producing warnings like:
Screen Shot 2021-08-20 at 11 29 00 AM
Screen Shot 2021-08-20 at 11 28 49 AM

Can you look into why that isn't running, or file an issue?  I wonder where else we are missing uninitialized variables.

LGTM for readability.

**Comment by chunhtai** [comment]
> Why didn't the original code produce a `Undefined or garbage value returned to caller` or an uninitialized static analyzer warning? That's concerning...
> 
> Should be producing warnings like:
> Screen Shot 2021-08-20 at 11 29 00 AM
> Screen Shot 2021-08-20 at 11 28 49 AM
> 
> Can you look into why that isn't running, or file an issue? I wonder where else we are missing uninitialized variables.
> 
> LGTM for readability.

Will ARC or MRC affect the linting? I heard ARC will default the nil if not set

**Comment by jmagman** [comment]
> > Why didn't the original code produce a `Undefined or garbage value returned to caller` or an uninitialized static analyzer warning? That's concerning...
> > Should be producing warnings like:
> > Screen Shot 2021-08-20 at 11 29 00 AM
> > Screen Shot 2021-08-20 at 11 28 49 AM
> > Can you look into why that isn't running, or file an issue? I wonder where else we are missing uninitialized variables.
> > LGTM for readability.
> 
> Will ARC or MRC affect the linting? I heard ARC will default the nil if not set

That example screenshot is MRC code.

**Pull request merged**